kubespawner/spawner.py

```diff
@@ -1969,6 +1969,18 @@ async def poll(self):
                             await self.stop(now=True)
                         return c["state"]["terminated"]["exitCode"]
                     break
+
+            # pod running. Check and update server url if it changed!
+            # only do this if fully running, not just starting up
+            if self.is_pod_running(pod):
+                pod_url = self._get_pod_url(pod)
+                if self.server.url != pod_url:
```

Comment:
It appears this can fail like this

![image](https://user-images.githubusercontent.com/3837114/109298925-934f1c80-7834-11eb-9da9-d9ef3fb09207.png)

Reply:
I suspect this is related to our mocking and not a 'real' possibility, but added an `if self.server` just to be safe, since it only makes sense to do this check when both are defined and can be compared.